# Inverted — a quantitative teardown 🔬
### Conditional forward returns by horizon · the strengthening gap · the variable-lead caveat

![Signal: Real](https://img.shields.io/badge/Signal-Real-2ea44f?style=flat-square)
![Tradability: Fragile](https://img.shields.io/badge/Tradability-Fragile-dab617?style=flat-square)
![Forecasts equity downturns?: Confirmed](https://img.shields.io/badge/Forecasts_equity_downturns%3F-Confirmed-8b949e?style=flat-square)

The deep companion to the [notebook for the curious](01_for_the_curious.ipynb). A real signal whose power lives at the 18–24-month horizon.

> ⚠️ **Not investment advice.** ^TNX/^IRX/^GSPC, monthly, 1985–2026 (Yahoo). Sources in [`docs/references.md`](../docs/references.md).

In [1]:
import sys, os
sys.path.insert(0, os.path.abspath("../../.."))  # repo root (quantlab/)
sys.path.insert(0, os.path.abspath(".."))        # study package (inverted/)
%matplotlib inline
import matplotlib.pyplot as plt
plt.rcParams["figure.figsize"] = (10, 5.5); plt.rcParams["axes.grid"] = True
import numpy as np, pandas as pd
from inverted import data, strategy as st
d = data.fetch_panel()                         # cache-first
slope = st.curve_slope(d["long_yield"], d["short_yield"])


## Verdict, up front

| Axis | Stamp | Why |
|---|---|---|
| Signal | **Real** | 18m +1% vs +16%; 24m −9% vs +22% |
| Tradability | **Fragile** | rare (~5%), 6–24m variable lead |
| Forecasts downturns? | **Confirmed** | the canonical recession barometer |

> 💡 *In plain words:* a real warning, a terrible sell button.

## 1 · The claim, steelmanned

- **H₁:** equity returns are lower after an inverted curve.
- **H₂:** the gap is economically large at the recession-lead horizon (18–24m).
- **H₃:** it's a usable *timer* (near-term, reliable).

## 2 · So what? — what rides on each

H₁/H₂ make it a real risk signal; H₃ decides whether you can trade it mechanically. A real-but-slow signal argues for posture, not timing.

## 3 · How we'd know — the protocol

Curve slope → forward S&P return after inverted vs normal at 12/18/24 months → the inverted share and the lead structure.

## 4 · The teardown

### 4.1 Conditional forward returns by horizon

In [2]:
import pandas as pd
display(pd.DataFrame({f'{h}m':st.conditional_forward(slope,d['eq'],h) for h in (12,18,24)}).T[['inverted_fwd','normal_fwd','gap','n_inverted']].round(3))

,inverted_fwd,normal_fwd,gap,n_inverted
12m,0.061,0.103,-0.042,30.0
18m,0.010,0.158,-0.148,24.0
24m,-0.087,0.217,-0.304,22.0


> 💡 *In plain words:* the gap widens from −4% (12m) to −15% (18m) to −30% (24m). **H₁, H₂ hold** strongly.

### 4.2 The variable lead — why H₃ fails

The 12-month gap is only −4.2%: in the year after an inversion, markets are usually still *up*. The damage comes at 18–24 months, but the exact timing varies recession to recession (6–24m). A mechanical 'sell on inversion' rule therefore sits out a year of gains and may be wrong-footed by the melt-up (1998, 2019, 2023). **H₃ rejected** — it's not a timer.

## 5 · The verdict

H₁/H₂ hold, H₃ rejected → Signal `REAL`, Tradability `FRAGILE`, the forecast `CONFIRMED`.

## 6 · Could you trade it?

As a risk-posture input (raise cash / trim beta when the odds rise), not a dated switch. Combine with a confirming trigger (e.g. price below its 200-day) to avoid the year-early exit.

## 7 · Going further

Forks: (a) the 10y−2y vs 10y−3m variants; (b) un-inversion (the *re-steepening* often marks the actual recession start); (c) curve + a trend filter as a combined timer. Backlog: [`docs/pwb_strategies_inventory.md`](../../../docs/pwb_strategies_inventory.md).